rust/protocol/src/curve/curve25519.rs

```diff
@@ -12,17 +23,34 @@ use sha2::{Digest, Sha512};
 use subtle::ConstantTimeEq;
 use x25519_dalek::{PublicKey, StaticSecret};
 
-const AGREEMENT_LENGTH: usize = 32;
+use std::fmt;
+
+/// Length of an agreed-upon key after a DH exchange.
+pub const AGREEMENT_LENGTH: usize = 32;
+/// Length of a private key.
 pub const PRIVATE_KEY_LENGTH: usize = 32;
+/// Length of a public key.
 pub const PUBLIC_KEY_LENGTH: usize = 32;
+/// Length of a signature.
 pub const SIGNATURE_LENGTH: usize = 64;
 
+/// A flexible key pair type wrapping [`PublicKey`] and [`StaticSecret`].
+///
+/// Operations here are simple enough to *not* produce fallible [`Result`]s. Instead we use static
+/// slice lengths to avoid having to handle any error cases.
 #[derive(Clone)]
 pub struct PrivateKey {
     secret: StaticSecret,
 }
 
+impl fmt::Debug for PrivateKey {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "PrivateKey {{ secret: {:?} }}", self.secret.to_bytes())
+    }
+}
+
 impl PrivateKey {
+    /// Generate a [`StaticSecret`] and then a [`PublicKey`] from a [`CryptoRng`].
```

Comment:
This is phrased in terms of the implementation. "Generate a new private key from a CryptoRng" is fine.